direct.go

```diff
@@ -286,32 +365,19 @@ func (d *direct) tlsConfig(m *Masquerade) *tls.Config {
 	return tlsConfig
 }
 
-// headCheck checks to make sure we can actually make a DDF head request through a
-// given masquerade. We don't reuse the underlying connection here because that confuses
-// the http.Client's internal transport.
-func (d *direct) headCheck(m *Masquerade) error {
-	trans := &http.Transport{
-		Dial: func(network, address string) (net.Conn, error) {
-			return d.dialServerWith(m)
+func httpTransport(conn net.Conn, clientSessionCache tls.ClientSessionCache) http.RoundTripper {
+	return &directTransport{
+		Transport: http.Transport{
+			Dial: func(network, addr string) (net.Conn, error) {
+				return conn, nil
+			},
+			TLSHandshakeTimeout: 40 * time.Second,
+			DisableKeepAlives:   true,
```

Comment:
Yeah we could kill `DisableKeepAlives` here and could just use `IdleConnTimeout`